core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java

```diff
@@ -896,6 +900,12 @@ public static boolean canCastFrom(
     if (toType.equals(fromType)) {
       return true;
     }
+    // If fromType is a measure, you should compare the inner type otherwise it will
+    // always return TRUE because the SqlTypeFamily of MEASURE is ANY
+    if (isMeasure(fromType)) {
```

Comment:
@tanclary If I understand this code correctly, this is testing for implicit (I think) conversions presumably as part of type checking on behalf of the caller. But it's not wholly clear to me that measures ought to be implicitly convertible to anything without applying `AGGREGATE`. Is it meaningful to compute anything with measures in themselves, outside an aggregation context? At the very least, there's an assumption of naked measures and implicit aggregation, right?

Reply:
Maybe you're right. This code allows the inner type to be compared to the toType.